crates/bevy_reflect/src/reflect.rs

```diff
@@ -62,7 +63,7 @@ pub unsafe trait Reflect: Any + Send + Sync {
     /// instance instead.
     ///
     /// [`TypeRegistry::get_type_info`]: crate::TypeRegistry::get_type_info
-    fn get_type_info(&self) -> TypeInfo;
+    fn get_type_info(&self) -> &'static TypeInfo;
```

Comment:
Thoughts on this change? I think it makes a lot more sense to return a static reference, but I understand that enforcing this on users could be controversial. My thought is that it's unlikely to be implemented directly, so it should be fine.

Reply:
Yeah, I'm fine with this.